### PR TITLE
Inset box shadow fixes

### LINKED ISSUE #183: Simple inset box shadows don't draw anything
```
<style>
  html, body {
    margin: 0;
  }
  .b {
    width: 300px;
    height: 20px;
    margin: 50px;
    background: green;
    box-shadow: inset 0 0 0 2px #0F0;
  }

</style>


  <div class="b">
  </div>
```

### PR DESCRIPTION
**Pull request by anderco**
Hi,

I found this project interesting and figured I try to contribute, so I looked at the open issues and tried to fix #183. I saw a couple more issues with inset box shadows, so here's an attempt to fix those.

I'm not exactly sure what's the best way to test these changes. I did manual testing, but pointers for a better method are appreciated.

Thanks,
Ander

### CODE REVIEW

**Comment by glennw** [comment]
Thanks! r? @pcwalton

File: src/batch_builder.rs [hunk 1]
```diff
@@ -1884,12 +1884,19 @@ impl BoxShadowMetrics {
...
-                               spread_radius: f32)
+                               spread_radius: f32,
+                               clip_mode: BoxShadowClipMode)
                                -> Rect<f32> {
     let mut rect = (*box_bounds).clone();
     rect.origin.x += box_offset.x;
     rect.origin.y += box_offset.y;
-    rect.inflate(spread_radius, spread_radius)
+
+    let _spread = match clip_mode {
```
Comment (pcwalton) [inline src/batch_builder.rs]: I would just make this:

    if clip_mode == BoxShadowClipMode::Inset {
        spread_radius = -spread_radius
    }

and change `spread_radius` in the function signature above to `mut spread_radius`.

**Comment by pcwalton** [comment]
Looks good other than that one nit. Thanks a lot!

File: src/batch_builder.rs (hunk 1 above)
Comment (anderco) [inline src/batch_builder.rs]: Makes sense. I'll update.

**Pull request merged**